notebooks/_06-Architecture.md

```diff
@@ -0,0 +1,429 @@
+---
+jupyter:
+  jupytext:
+    formats: ipynb,md
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.2'
+      jupytext_version: 1.3.3
+  kernelspec:
+    display_name: Python 3
+    language: python
+    name: python3
+---
+
+# Architecture
+
+
+## 1. At a “high” level, how will the project deliver each of the needed requirements?
+- **Try to think in terms of some manageable (e.g. 10-20) steps**
+- **List needed intermediate products (e.g. pseudo-code, unit tests, etc.)**
+- **Try to be concise but comprehensive?**
+
+The project is split into two tasks:
+
+The first task will be based on simulation work and will go as follows:
+
+For a causal graph where all Xs are independent:
+ - Ingest data
+ - Manipulate data as needed, the output of this step will be a cleaned dataset
+ - Specify a choice model based on the data at hand, the specification should follow required conventions by third party dependencies
+ - The cleaned dataset will be used in a function that simulates choice data based on the specified model, resulting in a dataset with simulated choices
+ - The dataset will be used to estimate a choice model and recover the necessary parameters. These parameters can be stored in a dictionary.
+ - The previous two steps will be repeated for N times (say 100) and the parameters from each iteration will be stored in a dictionary entry.
+ - A function that takes in the parameters of the estimated model and results in the percentage of time that the true parameter is recovered.
+
+
+For a realistic causal graph with confounding:
+ - Ingest data
+ - Manipulate data as needed, the output of this step will be a cleaned dataset
+ - Specify a choice model based on the data at hand, the specification should follow required conventions by third party dependencies
+ - Specify the realistic causal graph with desired confounding structures. The causal graph will be saved as a causalgraphicalmodels object and can be saved as .png file
+ - The cleaned dataset will be used in a function that simulates choice data based on the specified model, resulting in a dataset with simulated choices
+ - The dataset will be used to estimate a choice model and recover the necessary parameters. These parameters can be stored in a dictionary.
+ - The previous two steps will be repeated for N times (say 100) and the parameters from each iteration will be stored in a dictionary entry.
+ - A function that takes in the parameters of the estimated model and results in the percentage of time that the true parameter is recovered.
```

Comment:
See question above about how to deal with multivariate parameter vectors. How will we report the results given that we have more than one parameter being estimated?

Reply:
See my reply above:
 your suggestion sounds good to me!